src/Runner.Worker/ActionRunner.cs

```diff
@@ -285,16 +285,20 @@ public async Task RunAsync()
 
         }
 
-        public bool TryEvaluateDisplayName(DictionaryContextData contextData, IExecutionContext context)
+        /// <summary>
+        /// Attempts to evaluate the DisplayName of this IActionRunner.
+        /// Returns true if the DisplayName is already present or it was successfully evaluated.
+        /// </summary>
+        public bool TryEvaluateDisplayName(DictionaryContextData contextData, IExecutionContext context, out bool updated)
         {
             ArgUtil.NotNull(context, nameof(context));
             ArgUtil.NotNull(Action, nameof(Action));
 
-            // If we have already expanded the display name, there is no need to expand it again
-            // TODO: Remove the ShouldEvaluateDisplayName check and field post m158 deploy, we should do it by default once the server is updated
+            updated = false;
+            // If we have already expanded the display name, don't bother attempting [re-]expansion.
             if (_didFullyEvaluateDisplayName || !string.IsNullOrEmpty(Action.DisplayName))
             {
-                return false;
+                return true;
```

Comment:
This is a subtle change in behavior, but fortunately there weren't many existing callers (mainly just unit tests) and even fewer existing callers that even bothered checking the return value

Reply:
The return value is `IsDisplayNameSet`, and the out var says `IsUpdated[JustNow]`.

Why do we need both? 🤔 Is it to avoid setting it again in `StepsRunner.TryUpdateDisplayName()`?

Reply:
Consider code that invokes `EvaluateDisplayName` multiple times.

Example:
```
if (EvaluateDisplayName(this.ExecutionContext.ExpressionValues, this.ExecutionContext)) 
{
   this.ExecutionContext.UpdateTimelineRecordDisplayName(this.DisplayName);
}
   .
   .
   .
if (EvaluateDisplayName(this.ExecutionContext.ExpressionValues, this.ExecutionContext)) 
{
   this.ExecutionContext.UpdateTimelineRecordDisplayName(this.DisplayName);
}
```

Which TimelineRecordUpdate is the correct one?